openpype/hosts/maya/plugins/publish/extract_arnold_scene_source.py

```diff
@@ -0,0 +1,160 @@
+        mask = arnold.AI_NODE_ALL
+
+        node_types = {
+            "options": arnold.AI_NODE_OPTIONS,
+            "camera": arnold.AI_NODE_CAMERA,
+            "light": arnold.AI_NODE_LIGHT,
+            "shape": arnold.AI_NODE_SHAPE,
+            "shader": arnold.AI_NODE_SHADER,
+            "override": arnold.AI_NODE_OVERRIDE,
+            "driver": arnold.AI_NODE_DRIVER,
+            "filter": arnold.AI_NODE_FILTER,
+            "color_manager": arnold.AI_NODE_COLOR_MANAGER,
+            "operator": arnold.AI_NODE_OPERATOR
+        }
+
+        for key in node_types.keys():
+            if instance.data.get("mask" + key.title()):
+                mask = mask ^ node_types[key]
+
+        # Motion blur
+        attribute_data = {
+            "defaultArnoldRenderOptions.motion_blur_enable": instance.data.get(
+                "motionBlur", True
+            ),
+            "defaultArnoldRenderOptions.motion_steps": instance.data.get(
+                "motionBlurKeys", 2
+            ),
+            "defaultArnoldRenderOptions.motion_frames": instance.data.get(
+                "motionBlurLength", 0.5
+            )
+        }
+
+        # Write out .ass file
+        kwargs = {
+            "filename": file_path,
+            "startFrame": instance.data.get("frameStartHandle", 1),
+            "endFrame": instance.data.get("frameEndHandle", 1),
+            "frameStep": instance.data.get("step", 1),
+            "selected": True,
+            "asciiAss": self.asciiAss,
+            "shadowLinks": True,
+            "lightLinks": True,
+            "boundingBox": True,
+            "expandProcedurals": instance.data.get("expandProcedurals", False),
+            "camera": instance.data["camera"],
+            "mask": mask
+        }
+
+        filenames = self._extract(
+            instance.data["setMembers"], attribute_data, kwargs
+        )
+
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        representation = {
+            "name": "ass",
+            "ext": "ass",
+            "files": filenames if len(filenames) > 1 else filenames[0],
+            "stagingDir": staging_dir,
+            "frameStart": kwargs["startFrame"]
+        }
+
+        instance.data["representations"].append(representation)
+
+        self.log.info(
+            "Extracted instance {} to: {}".format(instance.name, staging_dir)
+        )
+
+        # Extract proxy.
+        if not instance.data.get("proxy", []):
+            return
+
+        kwargs["filename"] = file_path.replace(".ass", "_proxy.ass")
+        filenames = self._extract(
+            instance.data["proxy"], attribute_data, kwargs
+        )
+
+        representation = {
+            "name": "proxy",
+            "ext": "ass",
+            "files": filenames if len(filenames) > 1 else filenames[0],
+            "stagingDir": staging_dir,
+            "frameStart": kwargs["startFrame"],
+            "outputName": "proxy"
+        }
```

Comment:
The same name and ext in the second representation won't affect the publish of the first rep. And this may help the loader to load the published content and proxy. 
```
representation = {
            "name": "ass",
            "ext": "ass",
            "files": the_proxy_filepaths,
            "stagingDir": staging_dir,
            "outputName": "proxy"
}
instance.data["representations"].append(representation)
```

Reply:
Having the same `name` means the Loader will think that the proxy representation can be loaded, so we need to differentiate between the original and proxy.

Reply:
> Having the same `name` means the Loader will think that the proxy representation can be loaded, so we need to differentiate between the original and proxy.

oh i see. then it makes sense to have different `name`.